src/types/address.rs

```diff
@@ -353,4 +336,40 @@ mod tests {
         assert_eq!(address, de_addr);
     }
 
+    #[test]
+    fn from_bitcoin_address_to_address() {
+        let bitcoin_address: BitcoinAddress =
```

Comment:
You shouldn't need the explicit type hint if you call `BitcoinAddress::from_str`. Usually though, calling `parse()` is more convenient since you don't need to import `std::str::FromStr` for this. For `parse()` you will need the type-hint. Go with one or the other but no both :)

Reply:
you're right, the compiler is fine, but CLion was complaining with: ```expected `Address`, found `Address` more... ```
I remove the explicit types.